[ENH] Implementation Added: Enhanced filter_tags functionality

DebjyotiRay opened the pull request:
#### Reference Issues/PRs
Fixes #8210.

#### Changes:
- Added support for string and list inputs to filter_tags parameter
- String input is equivalent to {tag: True}
- List input is equivalent to {tag: True for tag in list}
- Added comprehensive tests for new functionality
- All tests pass successfully

#### What should a reviewer concentrate their feedback on?
Implementation logic, documentation updates, test coverage, error handling, and backward compatibility

#### Tests added:
- `test_all_estimators_filter_tags_str()`: Tests string equivalence
- `test_all_estimators_filter_tags_list()`: Tests list equivalence
- `test_all_estimators_filter_tags_invalid_type()`: Tests error handling

fkiraly reviewed (changes requested):
Thanks!

For the "list of strings" feature, as the issue says:

> This latter feature requires adding it to `scikit-base` `all_objects`.

This is the package,
https://github.com/sktime/skbase
i.e., it should be added there, upstream, not here.

The docstring of course should also be changed here.

DebjyotiRay commented:
Thanks for your feedback @fkiraly !

Changes to skbase are referenced here: https://github.com/sktime/skbase/pull/422
Changes here:
1. Removed local list implementation- No longer handles list conversion in sktime

2. Updated docstring to document only string functionality:
   - `filter_tags` now accepts `str` or `dict` (removed `list of str`)
   - Kept string explanation: "If filter_tags is a string, it is equivalent to {filter_tags: True}"
   - Removed list explanation and example

fkiraly commented:
Really great! I left some comments on https://github.com/sktime/skbase/pull/422.

I would suggest we finish that first, and then merge this PR with the docstring once 422 is in a release.
